gadgets/cses-status/src/lib.rs

```diff
@@ -1,24 +1,73 @@
 use std::sync::{Arc, Mutex};
 
 use cses_query::{Problem, ProblemStatus};
-use egui::{Color32, CursorIcon, RichText, Sense};
+use egui::{Color32, CursorIcon, RichText, Sense, TextEdit};
+use serde::{Deserialize, Serialize};
 use utils::{Gadget, GadgetFactory, MutexExt};
 
 mod cses_query;
 
 pub struct CSESStatusGadget {
-    problem_data: Arc<Mutex<Vec<Problem>>>,
+    problem_data: ProblemData,
+    network_runtime: utils::NetworkRuntime,
+    ask_for_id: bool,
+    input: String,
 }
 
+type ProblemData = Arc<Mutex<Vec<Problem>>>;
+
 pub struct CSESStatusGadgetFactory;
 
+#[derive(Serialize, Deserialize)]
+struct CSESStatusConfig {
+    #[serde(rename = "user-id")]
+    user_id: u64,
+}
+
+impl CSESStatusGadget {
+    fn ask_for_id(&mut self, ui: &mut egui::Ui, ctx: egui::Context) {
+        ui.horizontal(|ui| {
+            ui.label(RichText::new("Configure your CSES UID: "));
+
+            let text_color = self.input.parse::<u64>().err().map(|_| Color32::RED);
+
+            let text_input = TextEdit::singleline(&mut self.input).text_color_opt(text_color);
+            let response = ui.add(text_input);
+
+            if response.lost_focus() && ui.input().key_pressed(egui::Key::Enter) {
+                if let Ok(user_id) = self.input.parse::<u64>() {
+                    utils::update_config_for_gadget(self, CSESStatusConfig { user_id }).unwrap();
+                    self.network_runtime.spawn(get_problem_data(
+                        self.problem_data.clone(),
+                        ctx,
+                        user_id,
+                    ));
+                    self.ask_for_id = false;
+                } else {
+                    // Do nothing this frame, next one the text will become red
```

Comment:
Instead of an empty `else` block, why not just move the comment right before the `if let` ?

Reply:
Personally I like how this reads beter

Reply:
Sure, I don't mind it